python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -49,10 +49,10 @@ class AutoscalingConfigProducer:
     """
 
     def __init__(self, ray_cluster_name, ray_cluster_namespace):
-        self._headers, self._verify = node_provider.load_k8s_secrets()
-        self._ray_cr_url = node_provider.url_from_resource(
-            namespace=ray_cluster_namespace, path=f"rayclusters/{ray_cluster_name}"
```

Comment:
we were repeating the code by having a kubernetes http client then doing the same logic in another class, even though we were already using the node provider module